src/Emulator/Peripherals/Peripherals/Timers/OpenTitan_Timer.cs

```diff
@@ -0,0 +1,201 @@
+                }
+                else
+                {
+                    timerExpired0IRQ.Set(false);
+                }
+            };
+
+            var registersMap = new Dictionary<long, DoubleWordRegister>
+           {
+                {(long)Registers.Control, new DoubleWordRegister(this)
+                    .WithFlag(0, out timerActive0, name: "active_0", changeCallback: (_, value) =>
+                             {
+                                 string enable_message = value ? "Enabled" : "Disabled";
+                                 this.Log(LogLevel.Noisy, $"{enable_message} timer");
+                                 Enabled = value;
+                             })
+                     .WithIgnoredBits(1,31)},
+
+                {(long)Registers.Config0, new DoubleWordRegister(this)
+                    .WithValueField(0, 12,  name: "prescaler", writeCallback: (_, value) =>
+                             {
+                                 prescalerValue0 = value;
+                                 Divider = prescalerValue0 + 1;
+                                 this.Log(LogLevel.Noisy, $"Config0: Set prescaler to 0x{value:X}");
+                             }, valueProviderCallback: _ =>
+                             {
+                                return prescalerValue0;
+                             })
+                     .WithValueField(16, 8, name: "step", writeCallback: (_, value) =>
+                             {
+                                 stepValue0 = value;
+                                 this.Log(LogLevel.Noisy, $"Config0: Set step to 0x{value:X}");
+                             }, valueProviderCallback: _=>
+                             {
+                                return stepValue0;
+                             })},
+
+                {(long)Registers.TimerValueLower0, new DoubleWordRegister(this)
+                     .WithValueField(0, 32, name: "timer value lower", writeCallback: (_, value) =>
+                             {
+                                timerValue0 &= (ulong)(0xFFFFFFFF<<32);
+                                timerValue0 |= value;
+                                Value = timerValue0;
+                                this.Log(LogLevel.Noisy, $"Set timer0 value lower 0x{value:X} - TimerValue0: 0x{timerValue0:X}");
+                             }, valueProviderCallback: _ =>
+                             {
+                                this.Log(LogLevel.Noisy, $"Get timer0 value lower - TimerValue0: 0x{timerValue0:X}");
+                                return (uint)((ulong)Value & (ulong)0xFFFFFFFF);
+                             })},
+
+                 {(long)Registers.TimerValueUpper0, new DoubleWordRegister(this)
+                     .WithValueField(0, 32, name: "timer value upper", writeCallback: (_, value) =>
+                             {
+                                timerValue0 &= (ulong)0x00000000_FFFFFFFF;
```

Comment:
is there a need for keeping the `timerValue0` field at all? Can't you read the `Value` property to do the calculations?